src/resolvers/git.cr

```diff
@@ -100,12 +100,25 @@ module Shards
       "git"
     end
 
+    private KEY_HOSTS = {"www.github.com", "github.com", "www.bitbucket.com", "bitbucket.com", "www.gitlab.com", "gitlab.com"}
+
     def self.normalize_key_source(key : String, source : String) : {String, String}
       case key
       when "git"
-        {"git", source}
+        uri = URI.parse(source)
+        downcased_host = uri.host.try &.downcase
+        scheme = uri.scheme.try &.downcase
+        if scheme.in?("git", "http", "https") && downcased_host && downcased_host.in?(KEY_HOSTS)
+          uri.scheme = "https"
+          downcased_path = uri.path.downcase
+          uri.path = downcased_path.ends_with?(".git") ? downcased_path : "#{downcased_path}.git"
```

Comment:
Are you sure it works with all providers (and not only GitHub)?

Reply:
yes, when you use `bitbucket: org/repo` it adds the .git

Reply:
Yes, all three providers work like that.